server.go

```diff
@@ -2570,6 +2582,9 @@ func (r *Results) UnmarshalJSON(b []byte) error {
 // Error returns the first error from any statement.
 // Returns nil if no errors occurred on any statements.
 func (r *Results) Error() error {
+	if r.Err != nil {
```

Comment:
This was a bug that was causing `results.Error()` to return `nil` even though the top level had an error instantiated.